core/src/main/java/io/questdb/griffin/engine/functions/math/LeastNumericFunctionFactory.java

```diff
@@ -150,14 +157,17 @@ public ObjList<Function> getArgs() {
         @Override
         public double getDouble(Record rec) {
             double value = Double.MAX_VALUE;
+            boolean allAreNull = true;
```

Comment:
same as for the "long", logic is naive and overly branchy, simpler logic could be:

```java
           double value = args.getQuick(0).getDouble(rec);
            for (int i = 1; i < n; i++) {
                final double v = args.getQuick(i).getDouble(rec);
                if (!Numbers.isNull(v)) {
                    value = Math.max(value, v);
                }
            }
            return value;
 ```

Reply:
`value = Math.max(value, v);` implies that this suggestion should be applied to greatest(), not least(). If so, the loop is incorrect in case when the first argument returns `NaN` - in that case the suggested function will return `NaN` even if all other arguments are non-`NaN`.